### ISSUE TITLE
Apple Silicon macOS: WezTerm is not able to recognize some of the patched fonts, probably because of nerd-fonts intra-family PostScript name clash

### ISSUE THREAD

**Issue by poetaman** [ryanoasis/nerd-fonts#579]
#### 🗹 Requirements
- [x] A brief but descriptive _title_ of your issue
- [x] I have searched the [issues](https://github.com/ryanoasis/nerd-fonts/issues) for my issue and found nothing related and/or helpful
- [x] I have read or scanned the [FAQ](https://github.com/ryanoasis/nerd-fonts/wiki/FAQ-and-Troubleshooting)
- [x] I have read or scanned the [Wiki](https://github.com/ryanoasis/nerd-fonts/wiki)

#### 🎯 Subject of the issue

[WezTerm](https://github.com/wez/wezterm) terminal emulator is not able to recognize certain fonts generated by fontpatcher. I tried with Operator Mono patched (my own copy), and JetBrainsMono patched (from your repo). For instance, JetBrainsMono regular italic is not recognized. I suspect it has something to do with name-clash of postscript names with its other weights (from the same nerd fonts repo). Note: This is being discussed as a "warning" [in another ticket](https://github.com/ryanoasis/nerd-fonts/issues/542#issue-747747106), and I have an orthogonal issue on WezTerm to detect fonts by "Full Name" [here](https://github.com/wez/wezterm/issues/483) (but anyway nerd-fonts should fix its intra-family PostScript naming issues).

```lua
-- ~/.config/wezterm/wezterm.lua
local wezterm = require 'wezterm';

return {
  font = wezterm.font("JetBrains Mono Italic Nerd Font Complete"),
  -- or
  -- font= wezterm.font("JetBrainsMonoNerdFontComplete-Italic")
  font_size = 19.0,
}

```

#### 🔧 Your Setup

- JetBrainsMono regular italic
- [WezTerm](https://github.com/wez/wezterm) terminal emulator
- macOS BigSur on Apple Silicon M1 (ARM) Mac mini

#### ★ Optional
Open ``>>wezterm`` from another terminal emulator. It will print the following error in the parent terminal:
![Screen Shot 2021-02-12 at 11 09 45 PM](https://user-images.githubusercontent.com/71736629/107844367-90a9ed00-6d87-11eb-8c62-04dce8342019.png)

Look at the bottom left of the screenshot for the warning. There are no multiple copies on the system, the warning is because of "PostScript name" name-clash between the various italic weights of JetBrainsNerd fonts.
![Screen Shot 2021-02-12 at 10 48 53 PM](https://user-images.githubusercontent.com/71736629/107844212-6c99dc00-6d86-11eb-899f-672cbd9b3342.png)

**Comment by reportaman** [comment]
Locally, for now, I did a workaround till patcher fixes this. I changed following values in 'name' table > 'name' records to make it clean, and make WezTerm work seamlessly with JetBrainNerdFonts. In my current install, each font weight has its own unique "family name", with subfamily name set to "Regular". As an example, for Jet Brains Nerd Mono Extra Bold Italic font:
```
Macintosh Font Family name: JetBrainsMonoNerdEBI
Macintosh Font Subfamily name: Regular
Macintosh Postscript name: JetBrainsMonoNerdFontComplete-ExtraBoldItalic
Macintosh Preferred Family: JetBrainsMonoNerdEBI
Macintosh Preferred Subfamily: Regular
Microsoft Font Family name: JetBrainsMonoNerdEBI
Microsoft Font Subfamily name: Regular
Microsoft Postscript name: JetBrainsMonoNerdFontComplete-ExtraBoldItalic
Microsoft Preferred Family: JetBrainsMonoNerdEBI
Microsoft Preferred Subfamily: Regular
```

**Comment by Finii** [comment]
This will be fixed with `v3.0.0` when the new naming scheme comes into effect for all fonts.

It is already fixed for JetBrains Mono via b98c2a1921bcf5a0c510e6e709ec36b73d7714ec
See also #542

For your self-patching, give `--makegroups` as command line option to the patcher.
